src/main/java/com/karankumar/booksapi/model/BookSeries.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2021  Karan Kumar
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.karankumar.booksapi.model;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.NonNull;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import java.util.List;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class BookSeries {
```

Comment:
How come we have this intermediate class?

Reply:
I didn't understand this one...

Reply:
Ah, I appreciate it was vague.

Currently, we have `Book` with a one-to-many relationship to `BookSeries`. This, in turn, has a one-to-many relationship to `BookSeriesMapping`. Could we remove `BookSeriesMapping` and only have a one-to-many relationship between `Book` and `BookSeriesMapping`? Let me know if this still isn't clear

Reply:
We could do that, but that will result in multiple entries for a single `BookSeries` entity (assuming multiple `Book` per series, single `Book`<>`BookSeries` ). In that case, we won't be able to uniquely identify a `BookSeries`...

Reply:
I made a mistake in the previous comment, so I've updated that now.

I don't think I've quite understood this just yet. Will we still able to uniquely identify a `BookSeries` by its ID?

Reply:
I will comment with an ER diagram for this which should make it understandable. `BookSeriesMapping` has a Foreign Key relationship with `BookSeries` so one should be able to find by Book Series ID.

PS: In general I think we should have an ER diagram since its getting a bit complex.

Reply:
Thanks, that's a good suggestion. I'll create an issue for that

Reply:
You can checkout the ER diagram here:
[https://ibb.co/jzPKZgh](url)

Reply:
Unfortunately, I get a 404

Reply:
https://ibb.co/C1xBZvL

Reply:
Thanks, I'll take a look at this soon and get back to you shortly

Reply:
Apologies for the delay. I haven't forgotten about this